zk_prover/src/chips/less_than_vertical.rs

```diff
@@ -0,0 +1,204 @@
+}
+
+/// Config for the LtVertical chip.
+#[derive(Clone, Copy, Debug)]
+pub struct LtVerticalConfig<const N_BYTES: usize> {
+    /// Denotes the lt outcome. If lhs < rhs then lt == 1, otherwise lt == 0.
+    pub lt: Column<Advice>,
+    /// Denotes the bytes representation of the difference between lhs and rhs.
+    pub diff: Column<Advice>,
+    /// Denotes the range within which each byte should lie.
+    pub u8: Column<Fixed>,
+    /// Denotes the range within which both lhs and rhs lie.
+    pub range: Fp,
+}
+
+impl<const N_BYTES: usize> LtVerticalConfig<N_BYTES> {
+    /// Returns an expression that denotes whether lhs < rhs, or not.
+    pub fn is_lt(&self, meta: &mut VirtualCells<Fp>, rotation: Option<Rotation>) -> Expression<Fp> {
+        meta.query_advice(self.lt, rotation.unwrap_or_else(Rotation::cur))
+    }
+}
+
+/// Chip that compares lhs < rhs.
+#[derive(Clone, Debug)]
+pub struct LtVerticalChip<const N_BYTES: usize> {
+    config: LtVerticalConfig<N_BYTES>,
+}
+
+impl<const N_BYTES: usize> LtVerticalChip<N_BYTES> {
+    /// Configures the LtVertical chip.
+    pub fn configure(
+        meta: &mut ConstraintSystem<Fp>,
+        q_enable: impl FnOnce(&mut VirtualCells<'_, Fp>) -> Expression<Fp>,
+        lhs: impl FnOnce(&mut VirtualCells<Fp>) -> Expression<Fp>,
+        rhs: impl FnOnce(&mut VirtualCells<Fp>) -> Expression<Fp>,
+    ) -> LtVerticalConfig<N_BYTES> {
+        let lt = meta.advice_column();
+        let diff = meta.advice_column();
+        let range = pow_of_two(N_BYTES * 8);
+        let u8 = meta.fixed_column();
+
+        meta.create_gate("lt gate", |meta| {
+            let q_enable = q_enable(meta);
+            let lt = meta.query_advice(lt, Rotation::cur());
+
+            let mut diff_bytes = Vec::<Expression<Fp>>::new();
+
+            diff_bytes.push(meta.query_advice(diff, Rotation::cur()));
+            diff_bytes.push(meta.query_advice(diff, Rotation::next()));
+
+            for i in 2..N_BYTES {
+                diff_bytes.push(meta.query_advice(diff, Rotation(i as i32)));
+            }
+
+            let check_a =
+                lhs(meta) - rhs(meta) - expr_from_bytes(&diff_bytes) + (lt.clone() * range);
+
+            let check_b = bool_check(lt);
+
+            [check_a, check_b]
+                .into_iter()
+                .map(move |poly| q_enable.clone() * poly)
+        });
+
+        meta.annotate_lookup_any_column(u8, || "LOOKUP_u8");
+
+        meta.lookup_any("range check for u8", |meta| {
+            let u8_cell = meta.query_advice(diff, Rotation::cur());
+            let u8_range = meta.query_fixed(u8, Rotation::cur());
+            vec![(u8_cell, u8_range)]
+        });
+
+        meta.lookup_any("range check for u8", |meta| {
+            let u8_cell = meta.query_advice(diff, Rotation::next());
+            let u8_range = meta.query_fixed(u8, Rotation::cur());
+            vec![(u8_cell, u8_range)]
+        });
+
+        for i in 0..N_BYTES {
```

Comment:
Considered the previous comment, I think loop should work fine as it is and you don't need the two previous `meta.lookup_any` with `cur` and `next` rotations.